### PR TITLE
[IOTDB-1902] Fix data abnormally increase after shutting down IoTDB, changing data dirs and restarting IoTDB 

### PR DESCRIPTION
**Pull request by THUMarkLau**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionFileInfo.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+      // after splitting result should contain more than 5 objects
+      throw new RuntimeException(
+          String.format("Path %s cannot be parsed into file info", filepath));
+    }
+
+    return new CompactionFileInfo(
+        splittedPath[length - 4],
+        splittedPath[length - 3],
+        splittedPath[length - 2],
+        splittedPath[length - 5].equals("sequence"),
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionFileInfo.java:78]: make this a constant

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/sizetiered/SizeTieredCompactionRecoverTask.java [hunk 2]
```diff
@@ -119,39 +117,40 @@ public void doCompaction() {
               targetFile);
           // the target tsfile is crashed, it is not completed
           writer.close();
+          LOGGER.info(
+              "{}-{} [Compaction][Recover] target file {} is not completed, delete it",
+              fullStorageGroupName,
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/sizetiered/SizeTieredCompactionRecoverTask.java:122]: full storage group name -> logical sg + virtual sg

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/sizetiered/SizeTieredCompactionRecoverTask.java [hunk 3]
```diff
@@ -119,39 +117,40 @@ public void doCompaction() {
               targetFile);
           // the target tsfile is crashed, it is not completed
           writer.close();
+          LOGGER.info(
+              "{}-{} [Compaction][Recover] target file {} is not completed, delete it",
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/sizetiered/SizeTieredCompactionRecoverTask.java:121]:
```suggestion
              "{}-{} [Compaction][Recover] target file {} failed, delete it",
```

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/sizetiered/SizeTieredCompactionTask.java [hunk 4]
```diff
@@ -38,12 +38,11 @@
...
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.HashSet;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicInteger;
 
-import static org.apache.iotdb.db.engine.compaction.inner.utils.SizeTieredCompactionLogger.SOURCE_NAME;
-import static org.apache.iotdb.db.engine.compaction.inner.utils.SizeTieredCompactionLogger.TARGET_NAME;
+import static org.apache.iotdb.db.engine.compaction.inner.utils.SizeTieredCompactionLogger.SOURCE_INFO;
+import static org.apache.iotdb.db.engine.compaction.inner.utils.SizeTieredCompactionLogger.TARGET_INFO;
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/sizetiered/SizeTieredCompactionTask.java:45]: fullStorageGroupName -> logicalStorageGroupName + "-" + virtualStorageGroupName

better to separate into two parameters

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionFileInfo.java [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+ */
+package org.apache.iotdb.db.engine.compaction;
+
+import org.apache.iotdb.db.conf.IoTDBDescriptor;
+
+import java.io.File;
+
+/**
+ * This class record the logical information of files in compaction, which is used to locate file in
+ * disk. File information includes is file sequence, storage group name, virtual storage group id,
```
[the remark this answers is not in the document]
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionFileInfo.java:27]: resolve

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionFileInfo.java [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+ * time partition id, file name. <b>This class cannot be initialized directly.</b> We provide some
+ * static method to create instance of this class.
+ */
+public class CompactionFileInfo {
```
[the remark this answers is not in the document]
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionFileInfo.java:31]: resolve

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionFileInfo.java [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+  }
+
+  /**
+   * This method find the File object of current file by searching it in every data directory. If
+   * the file is not found, it will return null.
+   */
+  public File getFileFromDataDirs() {
+    String[] dataDirs = IoTDBDescriptor.getInstance().getConfig().getDataDirs();
+    String partialFileString =
+        (sequence ? "sequence" : "unsequence")
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionFileInfo.java:136]: use constant

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionFileInfo.java (hunk 1 above)
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionFileInfo.java:78]: resolve

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/sizetiered/SizeTieredCompactionRecoverTask.java (hunk 2 above)
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/sizetiered/SizeTieredCompactionRecoverTask.java:122]: resolve

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/sizetiered/SizeTieredCompactionTask.java (hunk 4 above)
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/sizetiered/SizeTieredCompactionTask.java:45]: resolve

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionFileInfo.java (hunk 7 above)
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionFileInfo.java]: resolve

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**